pkg/k8shandler/common.go

```diff
@@ -0,0 +1,491 @@
+package k8shandler
+
+import (
+	"fmt"
+	"github.com/sirupsen/logrus"
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"sort"
+	"strconv"
+
+	"github.com/openshift/elasticsearch-operator/pkg/utils"
+
+	api "github.com/openshift/elasticsearch-operator/pkg/apis/elasticsearch/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// addOwnerRefToObject appends the desired OwnerReference to the object
+func addOwnerRefToObject(o metav1.Object, r metav1.OwnerReference) {
+	if (metav1.OwnerReference{}) != r {
+		o.SetOwnerReferences(append(o.GetOwnerReferences(), r))
+	}
+}
+
+func getImage(commonImage string) string {
+	image := commonImage
+	if image == "" {
+		image = elasticsearchDefaultImage
+	}
+	return image
+}
+
+func getNodeRoleMap(node api.ElasticsearchNode) map[api.ElasticsearchNodeRole]bool {
```

Comment:
Not sure we wish to do here, but a better choice over map and bools may be a set: vendor/golang.org/x/text/internal/stringset/set.go

Reply:
this is for a x-descriptor, we are required to have a map so that we can show the different nodes in their states and correctly display them as counts, etc.